src/utils/verifiable-credential/baseVC.ts

```diff
@@ -49,44 +97,46 @@ export abstract class BaseVerifiableCredential {
     credentialSubject: CredentialSubject,
   ): SignTypedDataArgs {
     return {
-      domain: this.getEIP712TrustCredentialDomain(),
+      domain: this.getCredentialDomain(),
       message: this.getMessagePayload(
         this.getIssuerDid(address),
         credentialSubject,
       ),
-      types: this.credentialSubjectTypes,
+      types: {
+        ...this.eip712DomainType,
+        ...this.verifiableCredentialType,
+        ...this.credentialSubjectTypes,
+      },
       primaryType: this.primaryType,
     };
   }
 
-  protected getEIP712Proof(signArg: SignTypedDataArgs, signature: Hex) {
+  protected getSignedEIP712Proof(signArg: SignTypedDataArgs, signature: Hex) {
+    const createDate = new Date().toISOString();
+    const verifiableCredential =
+      this.verifiableCredentialType.VerifiableCredential.slice();
+    verifiableCredential.push({
+      name: 'proof',
+      type: 'Proof',
+    });
+
     return {
       type: 'EthereumEip712Signature2021',
       proofValue: signature,
       proofPurpose: 'assertionMethod',
+      verificationMethod: `${
+        (signArg.message as Assertion).issuer
+      }#blockchainAccountId`,
```

Comment:
Do you mean that `verificationMethod` here is issuer blockchainAccountId, which will be hash of DID of issuer?

Reply:
actually, not sure yet, still waiting K3L to reply us where it is correct or not ,as they are the one who validate the message